configure.ac

```diff
@@ -115,14 +116,21 @@ AM_CONDITIONAL(HAVE_GCC, test "x$GCC" = "xyes")
 
 dnl === Dependencies ==========================================================
 
-AC_CHECK_PROG([XAPIAN_CONFIG], [xapian-config-1.3], [xapian-config-1.3])
-AS_IF([test "$XAPIAN_CONFIG"],
-      [ found_xapian=yes ],
-      [ AC_MSG_ERROR(["*** No Xapian headers found; you need to install xapian-config ***"]) ])
-
-CXXFLAGS="$CXXFLAGS `$XAPIAN_CONFIG --cxxflags`"
-
-XAPIAN_LIBS="`$XAPIAN_CONFIG --ltlibs`"
+AC_ARG_WITH([xapian-config], [AS_HELP_STRING([--with-xapian-config], [The xapian-config binary])],
+            [xapian_config=$withval],
+            [xapian_config=auto])
+
+AS_IF([test "x$xapian_config" = xauto], [
+  PKG_CHECK_MODULES(XAPIAN, [xapian-core])
+  CXXFLAGS="$CXXFLAGS $XAPIAN_CFLAGS"
```

Comment:
an extra nicety might be to output info about which xapian you are compiling with, in the summary at the end of configure.ac.